python/lsst/obs/subaru/gen3/hsc/instrument.py

```diff
@@ -111,17 +128,59 @@ def getCamera(self):
+        """Return the brighter-fatter kernel for HSC as a `numpy.ndarray`.
 
         This is a temporary API that should go away once obs_ packages have
-        a standardized approach to writing versioned cameras to a Gen3 repo.
+        a standardized approach to writing versioned kernels to a Gen3 repo.
+        """
+        path = os.path.join(getPackageDir("obs_subaru"), "hsc", "brighter_fatter_kernel.pkl")
+        with open(path, "rb") as fd:
+            kernel = pickle.load(fd, encoding='latin1')  # encoding for pickle written with Python 2
+        return kernel
+
+    def writeCuratedCalibrations(self, butler):
+        """Write human-curated calibration Datasets to the given Butler with
+        the appropriate validity ranges.
+
+        This is a temporary API that should go away once obs_ packages have
+        a standardized approach to this problem.
         """
+
+        # Write cameraGeom.Camera, with an infinite validity range.
         datasetType = DatasetType("camera", ("Instrument", "ExposureRange"), "TablePersistableCamera")
         butler.registry.registerDatasetType(datasetType)
         camera = self.getCamera()
         butler.put(camera, datasetType,
                    instrument=self.getName(),
                    valid_first=datetime.min,
                    valid_last=datetime.max)
+
+        # Write brighter-fatter kernel, with an infinite validity range.
+        datasetType = DatasetType("bfKernel", ("Instrument", "ExposureRange"), "NumpyArray")
+        butler.registry.registerDatasetType(datasetType)
+        # Load and then put instead of just moving the file in part to ensure
+        # the version in-repo is written with Python 3 and does need
+        # `encoding='latin1'` to be read.
+        bfKernel = self.getBrighterFatterKernel()
+        butler.put(bfKernel, datasetType,
+                   instrument=self.getName(),
+                   valid_first=datetime.min,
+                   valid_last=datetime.max)
+
+        # Write defects with validity ranges taken from obs_subaru/hsc/defects
+        # (along with the defects themselves).
+        datasetType = DatasetType("defects", ("Instrument", "Detector", "ExposureRange"), "Catalog")
+        butler.registry.registerDatasetType(datasetType)
+        defectPath = os.path.join(getPackageDir("obs_subaru"), "hsc", "defects")
+        dbPath = os.path.join(defectPath, "defectRegistry.sqlite3")
+        db = sqlite3.connect(dbPath)
+        db.row_factory = sqlite3.Row
+        sql = "SELECT path, ccd, validStart, validEnd FROM defect"
+        with butler.transaction():
+            for row in db.execute(sql):
+                dataId = dict(instrument="HSC", detector=row["ccd"],
+                              valid_first=readDateTime(row["validStart"]),
+                              valid_last=readDateTime(row["validEnd"]))
+                ref = butler.registry.addDataset(datasetType, dataId, run=butler.run, recursive=True)
```

Comment:
Yes and no - Gen3 `Butler` has two concepts here, `run` and `collection`.  A `Butler` that can write has a `run`, and that `run` is always associated with a `collection` (in fact they use the same name), but the `collection` can also contain data from other runs, and by convention includes all of the datasets used as input to the run as well (much like Gen2 repos can sort-of-contain data from their parent repositories).  Once a dataset has been produced, it can be added to additional `collections` or even removed from its original `collection`, but it is always associated with the same `run`.